[SOAR-1766][SOAR-1705] - Microsoft atp fix trigger (#570)
rapid7/insightconnect-plugins: microsoft_atp/help.md

# Requirements

* Windows Defender Advanced Threat Protection application credentials

# Documentation

## Setup

This plugin uses the Windows Defender ATP API. It will use an Azure application to connect to the API and run 
actions from InsightConnect. 

> For information on how to setup your application and assign permssions go here:  

Reviewer: ```suggestion
For information on how to setup your application and assign permissions go here:  
```

https://docs.microsoft.com/en-us/windows/security/threat-protection/microsoft-defender-atp/exposed-apis-create-app-webapp

The connection configuration accepts the following parameters:

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|application_id|string|None|True|Application (client) ID|None|xxxxxxx-xxxxxx-xxxxx-xxxx|
|application_secret|credential_secret_key|None|True|Application secret|None|xxxxxxxxxxxxxxxxxxxxx|
|directory_id|string|None|True|Directory (tenant) ID|None|xxxx-xxxx-xxxx-xxxx-xxx|

Example input:

```
{
  "application_id": "xxxxxxx-xxxxxx-xxxxx-xxxx",
  "application_secret": "xxxxxxxxxxxxxxxxxxxxx",
  "directory_id": "xxxx-xxxx-xxxx-xxxx-xxx"
}
```

## Technical Details

### Actions

#### Get Machine Information

> This action is used to get details about a machine with it's ID.

Reviewer: ```suggestion
This action is used to get details about a machine from its ID.
```

      "investigationState":"Benign",
      "detectionSource":"WindowsDefenderAv",
      "category":"Malware",
      "title":"'EICAR_Test_File' malware was detected",
      "description":"Malware and unwanted software are undesirable applications that perform annoying, disruptive, or harmful actions on affected machines. Some of these undesirable applications can replicate and spread from one machine to another. Others are able to receive commands from remote attackers and perform activities associated with cyber attacks.\n\nThis detection might indicate that the malware was stopped from delivering its payload. However, it is prudent to check the machine for signs of infection.",
      "alertCreationTime":"2020-07-01T13:51:29.0741799Z",
      "firstEventTime":"2020-07-01T13:49:55.2853766Z",
      "lastEventTime":"2020-07-01T13:49:55.8520351Z",
      "lastUpdateTime":"2020-07-02T20:11:23.0966667Z",
      "resolvedTime":"2020-07-01T14:02:24.4812386Z",
      "machineId":"2df36d707c1ee508xyFf77f3dbfc95db65bc4a73",
      "computerDnsName":"example-desktop",
      "aadTenantId":"5c824599-ab2c-43ab-651x-3b886d4f8f10",
      "comments":[

      ],
      "evidence":[

>       ]

Reviewer 2: Collapse this JSON array if it's empty

}
```

#### Get Alerts Matching Key
#### Get Alerts

This trigger is used to get alerts that match a given key to its value.
This trigger is used to return all new alerts.

##### Input

|Name|Type|Default|Required|Description|Enum|
|----|----|-------|--------|-----------|----|
|key|string|None|True|The key to look for in the alert|None|
|value|string|None|True|The value to look for in the alert|None|
|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|frequency|integer|10|False|Poll frequency in seconds|None|10|

##### Output

|Name|Type|Required|Description|
|----|----|--------|-----------|
|results|[]alert|True|All alerts that contain the given key and match its value|

Example output:
Example input:

```
{

>    "results":[

Reviewer: Looks like the example output was removed

Author: I *think* this was fixed on regen.

      "investigationState":"Benign",
      "detectionSource":"WindowsDefenderAv",
      "category":"Malware",
      "title":"'EICAR_Test_File' malware was detected",
      "description":"Malware and unwanted software are undesirable applications that perform annoying, disruptive, or harmful actions on affected machines. Some of these undesirable applications can replicate and spread from one machine to another. Others are able to receive commands from remote attackers and perform activities associated with cyber attacks.\n\nThis detection might indicate that the malware was stopped from delivering its payload. However, it is prudent to check the machine for signs of infection.",
      "alertCreationTime":"2020-07-01T13:51:29.0741799Z",
      "firstEventTime":"2020-07-01T13:49:55.2853766Z",
      "lastEventTime":"2020-07-01T13:49:55.8520351Z",
      "lastUpdateTime":"2020-07-02T20:11:23.0966667Z",
      "resolvedTime":"2020-07-01T14:02:24.4812386Z",
      "machineId":"2df36d707c1ee508xyFf77f3dbfc95db65bc4a73",
      "computerDnsName":"example-desktop",
      "aadTenantId":"5c824599-ab2c-43ab-651x-3b886d4f8f10",
      "comments":[

      ],
      "evidence":[

>       ]

Reviewer 2: Same as above
